File: src/main/resources/public/model/GameState.js
var GameState = {
    spaces: {},
    winner: 0,
    currentPlayer: 1,
    player1Score: 0,
    player2Score: 0,
    create: function (loadedState) {
        if (loadedState) {
            GameState.spaces = loadedState.spaces;
            GameState.winner = loadedState.winner;
            GameState.currentPlayer = loadedState.currentPlayer;
            GameState.player1Score = loadedState.player1Score;
            GameState.player2Score = loadedState.player2Score;
        } else {
            GameState.spaces = Constants.defaultLayout;
            GameState.winner = 0;
            GameState.currentPlayer = 1;
            GameState.player1Score = 0;
            GameState.player2Score = 0;
        }
    },
    getMarbleAt: function (space) {
        return GameState.spaces[space];
    },
    removeMarble: function (space) {
        GameState.spaces[space] = 0;
    },
    setMarble: function (space, player) {
        GameState.spaces[space] = player;
    }
}

File: src/main/java/abalone/game/GameState.java
package abalone.game;

import abalone.dto.PlayerDto;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Map;
import java.util.Observable;
import java.util.UUID;

@EqualsAndHashCode(callSuper = true)
@Data
public class GameState extends Observable {

    private UUID id;
    private PlayerDto player1;
    private PlayerDto player2;
    private Map<String, Integer> spaces;
    private Integer winner = 0;
    private Integer currentPlayer = 1;
    private Integer player1Score = 0;
    private Integer player2Score = 0;

    public int getMarbleAtSpace(String space) {
        return spaces.get(space);
    }

    public void incrementPlayer1score() {
        player1Score++;
        setChanged();
    }

    public void incrementPlayer2score() {
        player2Score++;
        setChanged();
    }

    public void removeMarble(String space) {
        spaces.put(space, 0);
        setChanged();
    }

    public void setMarble(String space, int player) {
        spaces.put(space, player);
        setChanged();
    }
}
